src/asciidoc/core-beans.adoc

```diff
@@ -4562,14 +4562,16 @@ automatically resolved, with no special setup necessary.
 	}
 ----
 
-[NOTE]
+[WARNING]
 ====
-`@Autowired`, `@Inject`, `@Resource`, and `@Value` annotations are handled by Spring
+`@Autowired`, `@Inject`, `@Resource`, and `@Value`  annotations are handled by Spring
 `BeanPostProcessor` implementations which in turn means that you __cannot__ apply these
 annotations within your own `BeanPostProcessor` or `BeanFactoryPostProcessor` types (if
 any). These types must be 'wired up' explicitly via XML or using a Spring `@Bean` method.
-====
 
+Also, do not use `@Autowired`, `@Inject`, and `@Resource` annotations within a BeanFactory class nor any of its dependencies. Doing so can result in a `BeanCurrentlyInCreationException`. The `@Value` annotation may also cause the problem, d
+epending on whether it causes a bean to be instantiated.
+====
```

Comment:
I find the explanation at the bottom of https://github.com/spring-projects/spring-framework/commit/28e114b569d0e3e19ea0e0160704b6726810f428 to be insufficient. It should be a warning. It doesn't mention the consequences: getting an exception.

It's a comment in code, and I'm working on the reference guide. Also, it’s about FactoryBean, and I’m dealing with BeanFactory.  If we need to call out the connection between the two to address the issue of these annotations causing a synchronization problem that sometimes raises an exception, please let me know.

As for this particular content, suppose we borrow a bit of content from that code comment and make a warning that reads as follows: 

[WARNING]
Adding any of the `@Autowired`, `@Inject`, or `@Resource` annotations to a BeanFactory object or any of its dependencies can result in an `BeanCurrentlyInCreationException`, as the injections performed by those annotations may happen before the BeanFactory object has been completed.

I still need to know whether to include @Value in the list of annotations that can cause the problem.

Thanks.